src/components/ShareList.jsx

```diff
@@ -0,0 +1,60 @@
+import { useState } from 'react';
+import { shareList, useAuth } from '../api';
+import { useStateWithStorage } from '../utils';
+
+export function ShareList() {
+	const [emailData, setEmailData] = useState('');
+	const [listPath] = useStateWithStorage('tcl-shopping-list-path', null);
+
+	const { user } = useAuth();
+	const userId = user?.uid;
+	const userEmail = user?.email;
+
+	const handleTextChange = (event) => {
+		return setEmailData(event.target.value);
+	};
+
+	const shareCurrentList = async () => {
+		const listShared = await shareList(listPath, userId, emailData);
+
+		if (listShared === '!owner') {
+			alert('You cannot share the list you do not own.');
+		} else if (listShared === 'shared') {
+			alert('List was shared with recipient.');
+		} else {
+			alert(
+				"The list was not shared because the recipient's email address does not exist in the system.",
+			);
+		}
+	};
+
+	const handleEmailInputSubmit = (event) => {
+		event.preventDefault();
+
+		if (userEmail === emailData) {
+			alert('You cannot share the list with yourself.');
+		} else {
+			shareCurrentList();
```

Comment:
It seems like the promise returned from the `shareCurrentList` function is ignored. Perhaps we could add a `.then` and `setEmailData('')` when the promise resolves?